### DISCUSSION TITLE
Solar Energy Year!

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/Powerwall-Dashboard#217] [General]
I love seeing the full year of data on solar production at a glance.  In Los Angeles, we continue to have mostly cloudy days, but you can definitely tell we are headed into Spring (March 20 is the first day!).  

Here is my 8.5kW system over the past 12 months:

<img width="1058" alt="image" src="https://user-images.githubusercontent.com/836718/226255316-9a5d4564-a0c1-4c5d-a803-8abe9d5de4f4.png">
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/836718/226256239-090bcd5a-53ef-4097-b25b-e34d2cc4cb62.png">

My plans this year are to decrease our home energy usage to fit better within the solar production footprint.

How is your system doing?  Post your Solar Energy Year so we can all celebrate with you!  😀

**Comment by mcbirse** [comment]
Cool idea - is that 24 months or 12 months though for your Power Meters? (or is the 24 months a typo?)

Here is my 10kW system with 1x Powerwall over the past year (Australia/Sydney location):

![image](https://user-images.githubusercontent.com/108725631/226275829-916f6ae2-4ae6-4157-b265-96f27029b723.png)

![image](https://user-images.githubusercontent.com/108725631/226271629-a5bc0be9-f1b0-40c4-923d-159b3592f458.png)

Never thought to check this really, nice to see my Net Grid Usage. Winning. 😁

**Reply by jasonacox** [reply]
Typo! 😊  

So cool!!!  Love to see that @mcbirse !!  You are supplying nearly 6 MWh to your neighbors and you pulled less than a MWh for your own use... super green!  I'm going to try to follow you on your home usage metrics. 😉

**Reply by mcbirse** [reply]
So my figures were for a non-EV household.

However, I imagine that will be last time I will see such a great Net Grid Usage. I just picked up a Polestar 2 a few weeks ago, so my excess solar will now go to that instead of the grid. Still super green - no more burning petrol...!

**Reply by jasonacox** [reply]
100%!!!!  How do you like the Polestar? They look gorgeous.

**Reply by mcbirse** [reply]
Awesome car, very happy! 😀 

Agreed they look amazing - [here it is](https://bit.ly/407IBb8) just after getting ceramic coated and window tinting. It will probably never look that clean again, haha.

**Comment by dkerr64** [comment]
Before I installed Powerwall-Dashboard I was downloading data monthly and importing into Excel and creating the below graph.  I have not started to investigate but I imagine this would be pretty easy to recreate with Grafana.
![image](https://user-images.githubusercontent.com/17837612/226432077-c2904bfe-aa08-4046-8bfb-2e86d0f79c74.png)
What I like about this is that I am using a 30-day moving average to smooth out the daily fluctuations so I can quickly see when, on average, I generate more than I use (sholder seasons) and when I use more than I generate (winter when sun is low in the sky, summer when I am running air conditioning).

Once I had a full year of data I added in a 365-day moving average which clearly shows that I am generating just a little bit more than I use, on average, over a full year... and you can see that this is improving over time (I think mainly because we did a lot of post-covid travel in 2022 thus reducing our consumption)

I later added in a calendar month average (the green horizontals) which helps me compare to same month a year ago.

**Reply by jasonacox** [reply]
Whoa!  Nice David!  

What do your totals look like (or were you able to get the history data to import into the Dashboard)?

**Reply by dkerr64** [reply]
Yes I was able to import my history.  This is what my last 12 months look like.
![image](https://user-images.githubusercontent.com/17837612/226659082-a86164a4-36d9-4983-8016-0ca8dafaadaa.png)
I went extra-large at install and have 48 solar panels and 4 powerwalls.  My only regret is that I should have bought more powerwalls.  I am located in northeast US so solar generation varies a lot over the year.

**Reply by jasonacox** [reply]
That looks great @dkerr64 . I agree on the Powerwalls and wish I could have added another string.

**Reply by dkerr64** [reply]
I have managed to partially reproduce my Excel graph in Grafana.  But I cannot find a way to show a monthly average on this (like the horizontal green lines in my graph above)...
<img width="1733" alt="image" src="https://user-images.githubusercontent.com/17837612/227068136-8f022871-5a4d-4870-8012-fede1c19c3cb.png">

**Reply by jasonacox** [reply]
Nice @dkerr64 !  Can you post the query or panel code?

**Reply by dkerr64** [reply]
Do I just copy out the JSON from the panel.  Will that do it?

**Reply by dkerr64** [reply]
This is copy/paste of the JSON... 
[Solar-24-Month.json.zip](https://github.com/jasonacox/Powerwall-Dashboard/files/11067460/Solar-24-Month.json.zip)

**Reply by jasonacox** [reply]
When you export (share) a dashboard, select the "Export for sharing externally" toggle:

<img width="290" alt="image" src="https://user-images.githubusercontent.com/836718/227689944-1c0904dd-ca3f-4f9d-8a6d-7104175ae3ab.png">

Here it is for my system - very cool!

<img width="763" alt="image" src="https://user-images.githubusercontent.com/836718/227689776-b7d0c7ef-8051-42d4-bddf-ed68b88bdd38.png">

[SolarYear.json.zip](https://github.com/jasonacox/Powerwall-Dashboard/files/11067893/SolarYear.json.zip)

**Reply by ThePnuts** [reply]
This is great, I don't have enough data for a year yet though. Here is Jasons same file with a 2nd panel using Flux as well.
[SolarYear.json.zip](https://github.com/jasonacox/Powerwall-Dashboard/files/11068163/SolarYear.json.zip)

![image](https://user-images.githubusercontent.com/126296751/227700565-eef73ddc-e2aa-4428-bdb5-239a0faefeeb.png)

**Reply by ThePnuts** [reply]
I don't know if I can replicate your monthly averages with the totals on top of the line, but this is close. This is scaled down to 1y since I have less data.
Try 1:
![image](https://user-images.githubusercontent.com/126296751/227702573-660cb672-7776-41f7-b0c6-7aaf5f78c5a1.png)

Try 2:
![image](https://user-images.githubusercontent.com/126296751/227702646-d2df6c56-57df-46c4-8986-1dcd3a3f92c7.png)

**Reply by dkerr64** [reply]
@ThePnuts that is definite progress.  Can you share how you did that please?

**Reply by ThePnuts** [reply]
@jasonacox Well... its not what you think, I used the first example and then I turned it into a dashed line and figured out the width and all that and made it work using that. How ever, thanks to the time shift of a new day, it broke. workaround could be forcing this by month so the date doesnt shift each day, but I was working a different angle that I just got to tired on.

The other angle I have just forcing the date for 15 days for each month with the monthly average value. Here is the code for the 2nd attempt that is starting towards what I mentioned above. Its already shifting the data back 15 days so it can be centered on the month, I just need a loop in there or something to make it do how ever many days I want the line to be.

```
import "timezone"
import "date"

option location = timezone.location(name: "America/Los_Angeles")

from(bucket: "tesla_energy")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "kwh")
  |> filter(fn: (r) => r["_field"] == "solar")
  |> aggregateWindow(every: 1d, fn: sum, createEmpty: false)
  |> aggregateWindow(every: 1mo, fn: mean, createEmpty: false)
  |> map(fn: (r) => ({r with _field: "Monthly Average"}))
  |> map(fn: (r) => ({r with _time: date.sub(d: 15d, from: r._time)}))
```

**Reply by dkerr64** [reply]
@ThePnuts sorry but I am not familiar enough with Grafana to know how to take your example above and add it to my chart.  For example using the simply editor I cannot add both a sum() and a mean() aggregation... if I try and add mean it replaces sum.  Could you export the JSON for me please?

Thanks

**Reply by ThePnuts** [reply]
Yup, here ya go
[Year view-1679862032046.json.zip](https://github.com/jasonacox/Powerwall-Dashboard/files/11072981/Year.view-1679862032046.json.zip)

Can you do sql queries against Influxdb 1.8? could you do like a nested query first selecting for the day average, then take that to make the monthly average?

**Reply by dkerr64** [reply]
@ThePnuts are you pulling the data from a different db?  I only have the standard Powerwall-Dashboard install and even importing your json does not get the horizontal lines?  I must be missing something?

**Reply by ThePnuts** [reply]
@dkerr64 I am terribly sorry, when I responded it was late and I thought I was responding to Jason. I am not on the standard Powerwall-Dashboard. Long story short, I am using the latest version of influxdb 2+ which is not directly compatible to this project as it is. So I have been re-writing all of the queries from influxql to flux. I did the same to the query Jason shared and then expanded on it to include the monthly average like on your spreadsheet example.

Im not sure you could use my flux code with influx 1.8. Instead of bucket and measurement you would use something like powerwall/kWh but I really do not know sql much.

**Comment by ThePnuts** [comment]
Oooh nice. I didn't get PTO until the end of September, so anything before that was limited to usage/pw refill. Looking forward to my first full year.

11.2 kW system with 1 powerwall, Sacramento, CA area.

Still waiting for my first net negative week.

![image](https://user-images.githubusercontent.com/126296751/226492409-94cb4477-f596-432d-b900-8e78a2eeb417.png)
![image](https://user-images.githubusercontent.com/126296751/226492850-1c601342-4c19-4634-a53c-4ced2bc5fbac.png)

**Reply by jasonacox** [reply]
That's awesome @ThePnuts !  With a 11.2kW system, you're going to catch up quick this summer.  I bet that home usage spike in September was a hard working A/C fighting back our lovely California heat? 😅  I can relate. On my agenda for this year is to invest in a more efficient HVAC system.

**Reply by ThePnuts** [reply]
Oh yeah, gotta love the heat and its only increasing each year. I'm way way on the outskirts of Sac in more of a desert area, so we roll 95-105 from August to early October. I think that spike in September was about a week of 110-113.

We did a new HVAC 2 years ago, but I should have looked into it a lot more in preparation of solar. Hindsight I would have done a either a heat pump for the central air or a mini split system. I am still tempted to look into the mini split later since the primary rooms we use are on the same side of the house, an ideal situation for it. Right now my heat is on gas and we have cold winters.

**Reply by jasonacox** [reply]
> so we roll 95-105 from August to early October. I think that spike in September was about a week of 110-113.

Yikes! I started looking at the high-efficiency HVAC units with a variable-speed compressors.  Not cheap, but the math works out over time and I'm itching to get my "100% Solar Power" badge... or at least a bit closer to what @mcbirse has. 🥇 😉

**Comment by jgleigh** [comment]
16.3 kW system in Arizona.

![Screenshot 2023-03-23 at 8 02 24 AM](https://user-images.githubusercontent.com/654130/227245554-00d7b597-06e5-4e76-835b-173cb4528d78.png)

![Screenshot 2023-03-23 at 8 02 04 AM](https://user-images.githubusercontent.com/654130/227245571-76d7d040-f52e-40e9-8cc1-376aa7dfff33.png)

**Reply by dkerr64** [reply]
WOW! you sure do get a lot of sunshine in Arizona.

**Reply by jasonacox** [reply]
Wow is right!  Amazing @dkerr64 !

**Comment by cwagz** [comment]
Here is my year on our 12.3kW system. I was mining ethereum (about 900 watts continuous) up until it wasn't possible anymore in September 2022. I expect my usage to be less in 2023. I am really happy with how the system has worked out. We had a credit of -$41 on our first 12-month true up bill that just came last week.

I also received $120 from Tesla for power we sold during the 8 Virtual Power Plant events in 2022.

<img width="1255" alt="Screenshot 2023-03-26 at 4 23 34 PM" src="https://user-images.githubusercontent.com/13752647/227811554-ca93f930-b88f-427b-9efa-ad3e6b319f91.png">

<img width="1257" alt="Screenshot 2023-03-26 at 4 22 17 PM" src="https://user-images.githubusercontent.com/13752647/227811565-ad10de6a-728f-4277-97b7-173277b813d9.png">

**Reply by jasonacox** [reply]
Awesome! Looks great @cwagz!  Before getting PTO, I had miners fire up to chew on all the excess solar power we couldn't push to the grid. I would just cringe to see the powerwalls at 100% and the inverter throttled down from 6kW to 1kW for home use only. 😁

**Comment by jgleigh** [comment]
April is fun!
<img width="704" alt="Screenshot 2023-04-27 at 9 25 04 PM" src="https://user-images.githubusercontent.com/654130/235054442-e3547b1a-2aed-4944-93f2-50349d0bf7d8.png">

**Comment by JonMurphy** [comment]
Sorry for bumping this three year old thread!  I am hoping you can tell me how this was done.  I am always stuck at 1 month. 

Please post your query.

<img width="1561" height="1238" alt="Screenshot 2026-05-28 at 4 15 50 PM" src="https://github.com/user-attachments/assets/7bba330c-9abd-4b8b-952d-34806e164620" />

**Reply by jgleigh** [reply]
There's a Solar Energy Year panel further down the dashboard.

**Reply by JonMurphy** [reply]
Yes, I just found that!  I should have worded my request differently.  

I was looking to change either panel to my time period.  So as an example: all solar for 2025.  But it seems like each panel is locked into the blue :clock1: Last 1 month or :clock1: Last 1 year.  Hope that made sense...

**Reply by jasonacox-sam** [reply]
Hey @JonMurphy — welcome to the thread, no need to apologize for the bump! Great questions.

To change any panel's time range beyond the dashboard's global time picker:

1. **Edit the panel** → click the panel title → **Edit**
2. Go to the **Query options** section (right sidebar)
3. Set **Time range** override — e.g. `now-12M/M` for the last 12 months aligned to month boundaries, or `now-1y/y` for last calendar year

For the monthly bar chart look that @dkerr64 and others posted above, you'd want to set the panel's **Relative time** to something like `now-12M` and group your InfluxDB query by `time(1M)`.

If you want all panels on a row to share a longer range, you can also use **Dashboard links** or **Variables** — but per-panel overrides are the simplest starting point.

The built-in **Solar Energy Year** panel already does a lot of this, so you could also duplicate that panel and customize the query to show exactly what you want. 👍

— Sam

**Reply by JonMurphy** [reply]
> Go to the Query options section (right sidebar)

There is no Query or Query options in the right sidebar

**Reply by JonMurphy** [reply]
When in the **Edit** mode, below the graph, there is a **Queries** tab and to the right of the first **Data source** there is **Query options**.  

<img width="700" alt="Screenshot 2026-05-29 at 11 47 53 AM" src="https://github.com/user-attachments/assets/402eb2c3-4e12-4e98-8a52-1f4cd594d51e" />

<p>

> Set Time range override — e.g. now-12M/M for the last 12 months aligned to month boundaries, or now-1y/y for last calendar year

But there is no **Time range** override

**Reply by JonMurphy** [reply]
If I open **Query options** and remove the **Relative time** entry of **12M**, it auto changes to **1h**.  That allows the graph to use the time series in the top right corner.

<img width="440" height="253" alt="Screenshot 2026-05-29 at 12 07 02 PM" src="https://github.com/user-attachments/assets/e3e62463-7fe6-4a72-8a48-5adce859ffc6" />

<p>

and then I see what I am looking for!

<img width="2228" height="806" alt="Screenshot 2026-05-29 at 12 02 41 PM" src="https://github.com/user-attachments/assets/d1616ac1-f509-4d55-946d-4147a66f97e7" />

**Comment by jasonacox-sam** [comment]
Hey @JonMurphy — welcome! The yearly views you see in the earlier comments are just the standard Powerwall-Dashboard Grafana panels with the **time range expanded**.

To see your yearly data:

1. Open your Grafana dashboard (usually `http://your-pi-ip:9000`)
2. Click the **time range picker** in the top-right corner (it probably says "Last 30 days" or similar)
3. Change it to **"Last 1 year"** or set a custom absolute range (e.g., Jan 1 – Dec 31)

The InfluxDB queries in the dashboard automatically aggregate over whatever time range you select — no query changes needed.

**One caveat:** if you've only been running the dashboard for a month, you'll only have a month of data. The yearly charts in this thread are from users who've been collecting data for 12+ months. The data accumulates over time in InfluxDB.

If you want to backfill historical data, check out the `tesla-history.py` tool included with the dashboard — it can pull historical data from Tesla's cloud API to fill in gaps from before you started monitoring.

Hope that helps! 📊

— Sam
